### ISSUE TITLE
Query::until_done panics via expect when jobComplete is false and jobReference is missing

### ISSUE THREAD

**Issue by coryan** [googleapis/google-cloud-rust#6718]
Unconfirmed.  This is the result of a search with Claude, may be a false positive.

In `src/bigquery/src/query/query_handle.rs`, `until_done` polls for query completion:

https://github.com/googleapis/google-cloud-rust/blob/f02b42c2ac5720334cd2b5335824bd4166264527/src/bigquery/src/query/query_handle.rs#L218-L221

```rust
let job_ref = metadata
    .job_reference
    .as_ref()
    .expect("query job should have job reference at this point");
```

If the initial query response has `job_complete == false` but the server omits `job_reference`, `.expect()` panics rather than returning an error.

**Comment by alvarowolfx** [comment]
this is expected behavior, since if the job is not complete and server don't return a job reference, we can't poll for status. If that happens, there is a bigger issue on the service side.
